src/main/java/com/google/cloud/tools/app/module/DeleteAction.java

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.tools.app.module;
+
+import com.google.cloud.tools.app.Action;
+import com.google.cloud.tools.app.GCloudExecutionException;
+import com.google.cloud.tools.app.Option;
+import com.google.cloud.tools.app.ProcessCaller;
+import com.google.cloud.tools.app.ProcessCaller.ProcessCallerFactory;
+import com.google.cloud.tools.app.ProcessCaller.Tool;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * Deletes a version of one or more modules.
+ */
+public class DeleteAction extends Action {
+
+  private static ProcessCallerFactory processCallerFactory = ProcessCaller.getFactory();
+  private Collection<String> modules = UNSET_COLLECTION;
+  private String version = UNSET_STRING;
+  private String server = UNSET_STRING;
+
+  public static DeleteAction newDeleteAction() {
+    return new DeleteAction();
+  }
+
+  public DeleteAction setModules(Collection<String> modules) {
+    Preconditions.checkArgument(this.modules == UNSET_COLLECTION, "Modules can only be set once.");
+    this.modules = modules;
+    return this;
+  }
+
+  public DeleteAction setVersion(String version) {
+    Preconditions.checkArgument(this.version == UNSET_STRING, "Version can only be set once.");
+    this.version = version;
+    return this;
+  }
+
+  public DeleteAction setServer(String server) {
+    Preconditions.checkArgument(this.server == UNSET_STRING, "Server can only be set once.");
+    this.server = server;
+    return this;
+  }
+
+  @Override
+  public boolean execute() throws GCloudExecutionException {
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(version));
+
+    List<String> arguments = new ArrayList<>();
+    arguments.add("modules");
+    arguments.add("delete");
+    if (modules == null) {
+      arguments.add("default");
```

Comment:
I think so.. what do you think?

On Wed, Mar 23, 2016 at 1:40 PM, João Martins <notifications@github.com>
wrote:

> In src/main/java/com/google/cloud/tools/app/module/DeleteAction.java
> <https://github.com/GoogleCloudPlatform/gcp-tools-lib-for-java/pull/1#discussion_r57203717>
> :
>
> > +
> > +  public DeleteAction setServer(String server) {
> > +    Preconditions.checkArgument(this.server == UNSET_STRING, "Server can only be set once.");
> > +    this.server = server;
> > +    return this;
> > +  }
> > +
> > +  @Override
> > +  public boolean execute() throws GCloudExecutionException {
> > +    Preconditions.checkArgument(!Strings.isNullOrEmpty(version));
> > +
> > +    List<String> arguments = new ArrayList<>();
> > +    arguments.add("modules");
> > +    arguments.add("delete");
> > +    if (modules == null) {
> > +      arguments.add("default");
>
> Providing required parameters in the builder method would look
> ugly/unorthodox, but certainly has its advantages, like enforcing required
> parameters are provided at compile time.
>
> I'll make this change.
>
> For non-destructive operations, should we consider modules optional?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/gcp-tools-lib-for-java/pull/1/files/6ac1b655470fd1c096609594f40b529e7a82ad4c#r57203717>
>